tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -254,28 +379,65 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
 					tcpDumpNet1CMD)
 
 				By("Define and create a server container")
-				serverPod := createServerTestPod(serverNameDot1q, srIovNetworkDot1Q, nadCVLAN,
-					workerNodeList[1].Definition.Name, []string{tsparams.ServerIPv4IPAddress, tsparams.ServerIPv6IPAddress})
+				annotation := defineNetworkAnnotation(srIovNetworkDot1Q, nadCVLAN100, true)
+				serverPod := createServerTestPod(serverNameDot1q, workerNodeList[1].Definition.Name, testCmdNet2,
+					annotation)
 
-				By("Define and create a client container")
-				clientPod := createClientTestPod(clientNameDot1q, srIovNetworkDot1Q, nadCVLAN,
-					workerNodeList[0].Definition.Name, []string{tsparams.ClientIPv4IPAddress, tsparams.ClientIPv6IPAddress})
+				By("Define and create a 802.1Q client container")
+				annotation = defineNetworkAnnotation(srIovNetworkDot1Q, nadCVLAN100, false)
+				clientPod := createClientTestPod(clientNameDot1q, workerNodeList[0].Definition.Name, annotation)
 
 				By("Validate IPv4 and IPv6 connectivity between the containers over the qinq tunnel.")
-				err := cmd.ICMPConnectivityCheck(serverPod, []string{tsparams.ClientIPv4IPAddress,
-					tsparams.ClientIPv6IPAddress}, "net2")
+				err := cmd.ICMPConnectivityCheck(serverPod, clientIPAddressesNet2, intNet2)
 				Expect(err).ToNot(HaveOccurred(),
 					"Failed to ping the client container over the 802.1q connection.")
 
-				By("Validate IPv4 tcp traffic and dot1ad encapsulation from the client to server.")
-				validateTCPTraffic(clientPod, serverIPV4IP.String())
-
-				By("Validate IPv6 tcp traffic and dot1ad encapsulation from the client to server.")
-				validateTCPTraffic(clientPod, serverIPV6IP.String())
+				By("Validate IPv4 and IPv6 tcp traffic and dot1q encapsulation from the client to server")
+				validateTCPTraffic(clientPod, intNet2, serverIPAddressesNet2)
 
 				By("Validate that the TCP traffic is double tagged")
 				readAndValidateTCPDump(tcpDumpContainer, tcpDumpReadFileCMD, tcpDumpDot1QOutput)
 			})
+
+		It("Verify network traffic over a double tagging QinQ tunnel with multiple C-VLANs using the same S-VLAN",
```

Comment:
double tagging QinQ is too general - it's better 802.1q/802.1q or two times 802.1q

Reply:
How about double tagged 802.1Q ?